certora/specs/SafeWebAuthnSignerSingleton.spec

```diff
@@ -56,42 +97,44 @@ rule verifyIsValidSignatureAreEqual(env e){
     bytes data;
     bytes first_signature;
 
-    bool hashed_data_verified = verifySignatureHarnessed(e, keccak256(data), first_signature);
+    bytes4 magicValue_hashed = isValidSignature(e, data, first_signature);
 
-    bytes second_signature;
     bytes32 message;
-    bool message_verified = verifySignatureHarnessed(e, message, second_signature);
+    bytes4 magicValue_message = isValidSignature(e, message, first_signature);
 
-    assert (hashed_data_verified == true && message_verified == true) => message == keccak256(data);
-    assert message == keccak256(data) => hashed_data_verified == message_verified;
+    assert (magicValue_hashed == to_bytes4(0x20c13b0b) && magicValue_message == to_bytes4(0x1626ba7e)) => message == keccak256(data);
+    assert message == keccak256(data) => (magicValue_hashed == to_bytes4(0x20c13b0b) && magicValue_message == to_bytes4(0x1626ba7e)) || 
+                                         (magicValue_hashed != to_bytes4(0x20c13b0b) && magicValue_message != to_bytes4(0x1626ba7e));
 }
 
-
 /*
 ┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
 │ Once signer passed isValidSignature it will never fail on it after any call                                         │
 └─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
 */
 
 rule verifyIsValidSignatureWillContinueToSucceed(env e){
+    method f;
+    calldataarg args;
     bytes32 message;
     bytes signature;
 
     bool first_verified = verifySignatureHarnessed(e, message, signature);
     require first_verified == true;
 
+    f(e, args);
+
```

Comment:
We can strengthen the rule by allowing `verifySignatureHarnessed` to revert and checking that both before and after did not revert.

Also, instead of `require A; assert B;`, we can use `A => B`